modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileSetManager.java

```diff
@@ -82,13 +92,39 @@ protected DbaseFileReader openDbfReader(boolean indexed) throws IOException {
             return null;
         }
 
+        Charset charset = store.getCharset();
+
+        if (store.isTryCpgFile()
+                && shpFiles.get(CPG) != null
+                && (!shpFiles.isLocal() || (shpFiles.isLocal() && shpFiles.exists(CPG)))) {
+            try (BufferedReader br =
+                    new BufferedReader(
+                            new InputStreamReader(
+                                    shpFiles.getInputStream(CPG, this), ISO_8859_1))) {
+                String charsetName;
+                if ((charsetName = br.readLine()) != null) {
+                    try {
+                        charset = Charset.forName(charsetName.trim());
```

Comment:
Ok, so here you're taking a gamble, and hoping the codepage has the same name as a Java charset... I guess they will not always line up, but better than nothing, right? I quickly searched for an existing source of cpg->charset mappings but could not find one. Well, it's called "try CPG" after all :-D

Reply:
I was thinking about that, did some research, not very deep though, and didn't find any relevant information. `Charset` class itself is smart enough to deals with a number of charset's names and aliases, as javadoc states. If someone will report an exotic mapping we could add it later. ;-)